manifest/spark-operator.yaml

```diff
@@ -44,5 +47,9 @@ spec:
         image: gcr.io/spark-operator/spark-operator:v2.3.0-v1alpha1-latest
         imagePullPolicy: Always
         command: ["/usr/bin/spark-operator"]
+        ports:
+          - containerPort: 10254
         args:
         - -logtostderr
+        - -enable-metrics=true
```

Comment:
What happens when I don't use prometheus?

Reply:
Right now this only supports Prometheus. Let me know what else you believe we should support ? 
I can generalize this a little more by adding wrapper interfaces for each Prometheus constructs like Counter/Gauge so that we can add other implementations going further.  Some of the operator arguments like metrics-endpoint etc will still be specific to Prometheus and we can rename them appropriately. 

@mrow4a/@liyinan926  let me know what you guys think about this.